checkbox-ng/checkbox_ng/resume_menu.py

```diff
@@ -0,0 +1,343 @@
+            urwid.SolidFill(" "),
+            align="center",
+            valign="middle",
+            width=("relative", 100),
+            height=("relative", 100),
+        )
+        return urwid.Frame(urwid.AttrWrap(olay, "body"))
+
+    def _create_action_view(self):
+        """Create a view for the action selection menu."""
+        self._action_buttons = [
+            # label that goes on the button and the action string recorded
+            ("Add comment", "comment"),
+            ("Resume and skip the job", "skip"),
+            ("Mark the job as passed and continue", "pass"),
+            ("Mark the job as failed and continue", "fail"),
+            ("Resume and run the job again.", "rerun"),
+        ]
+        action_listbox_content = self._ACTION_MENU_STATIC_ELEMENTS + [
+            urwid.Button(btn[0]) for btn in self._action_buttons
+        ]
+
+        self._action_menu = urwid.LineBox(
+            urwid.ListBox(urwid.SimpleFocusListWalker(action_listbox_content))
+        )
+        self._action_view = urwid.AttrWrap(
+            urwid.Overlay(
+                self._action_menu,
+                self._body,
+                "center",
+                ("relative", 50),
+                "middle",
+                ("relative", 50),
+                left=10,
+            ),
+            "body",
+        )
+
+    def _create_comment_view(self):
+        """Create a view for the comment entry."""
+        self._comment_edit_box = urwid.Edit("Enter comment:\n")
+        box = urwid.LineBox(
+            urwid.Filler(
+                self._comment_edit_box,
+            )
+        )
+        self._comment_view = urwid.AttrMap(
+            urwid.Overlay(
+                box,
+                self._action_view,
+                "center",
+                ("relative", 45),
+                "middle",
+                ("relative", 30),
+                left=10,
+            ),
+            "body",
+        )
+
+    def run(self):
+        """
+        Run the menu and return the chosen session and action.
+        """
+        self.loop = urwid.MainLoop(
+            self._body,
+            palette,
+            unhandled_input=self._unhandled_input,
+            handle_mouse=False,
+        )
+        self.loop.run()
+        return ResumeOutcome(
+            self._chosen_session, self._chosen_action, self._comment
+        )
+
+    def _handle_input_on_session_menu(self, key):
+        if key == "enter":
+            chosen_session = self._entries[self.focused_index][0]
+            if chosen_session != self._chosen_session:
+                # operator may have selected a session and then written a
+                # comment but then went back to the previous menu, and
+                # selected another session, so we need to disregard the
+                # comment they entered
+                self._comment = ""
+                self._comment_edit_box.edit_text = ""
+                self._chosen_session = chosen_session
```

Comment:
There is a `@chosen_session.setter` above which comment indicates it's doing the same thing as this. Are both needed?

Reply:
No, they're not - good catch, TY.
Fixed in 45326ca26516b397c12c03b3c79c5b8cf40e7457